[EASI-4332] Add/edit/remove GRB reviewer

aterstriep opened the pull request:
# EASI-4332

[Figma](https://www.figma.com/design/G0UN35V7SY0g9nkTIESxsY/EASi-GRB-Experience?node-id=2266-44308&t=ZfZWTDvMFZ75OFxk-0)

## Description
- Form for adding and editing GRB reviewers
- Remove GRB reviewer functionality
- List of reviewers with edit/remove buttons on GRB review tab for testing purposes (table will be completed in separate ticket)
- Unit tests

## How to test this change
- Log in as GRT admin
- Go to GRB review tab of intake request (/governance-review-team/a2fa0d4b-909f-45d8-ad8c-90f22cf0db19/grb-review)
- Click "Add a GRB reviewer" button
- Complete form to add reviewer
- Click "edit" button under reviewer name
- Update role fields and submit form to edit reviewer
- Click "remove" button under reviewer name to test remove functionality


## PR Author Checklist
<!--
REQUIRED
    Ensure that each of the following is true before you submit this PR (or before it leaves "draft" status), and check each box to confirm
-->

- [ ] I have provided a detailed description of the changes in this PR.
- [ ] I have provided clear instructions on how to test the changes in this PR.
- [ ] I have updated tests or written new tests as appropriate in this PR.


## PR Reviewer Guidelines
<!--
This is just some static content to ensure we're following best practices when reviewing.
There is no need to edit this section.
-->
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- When approving a PR, provide a reason _why_ you're approving it
  - e.g. "Approving because I tested it locally and all functionality works as expected"
  - e.g. "Approving because the change is simple and matches the Figma design"
- Don't be afraid to leave comments or ask questions, especially if you don't understand why something was done! (This is often a great time to suggest code comments or documentation updates)
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.

ClayBenson94 commented on src/queries/SystemIntakeGRBReviewerQueries.ts:
Could you try and build these using the new GraphQL codegen tooling?

I think it should just be a matter of adding queries (and following the example) in https://github.com/CMS-Enterprise/easi-app/tree/main/src/gql/apolloGQL and using the new generated hooks instead of `useQuery`/`useMutation`

aterstriep commented:
@adamodd unit tests and lint issues have been fixed - this is fully ready for review.

patrickseguraoddball commented:
Just a small adjustment, the alerts are supposed to span grid-12

![Screenshot 2024-08-02 at 9 25 11 AM](https://github.com/user-attachments/assets/6a7f3804-9510-48f1-9bbd-12b92e14b591)

patrickseguraoddball commented:
Might want to check with UX if this order is supposed to match Figma

![Screenshot 2024-08-02 at 9 26 03 AM](https://github.com/user-attachments/assets/c70cd8e5-62a8-47e4-aae1-a2d1f9e26739)

patrickseguraoddball reviewed (changes requested):
Looks good,!  Just a few minor UX/style things.  Code-wise looks good :) 

After attempting to add the same user twice, I'm able to move any of the options back to '- Select -' and the submit button is enabled. Not a huge deal as I don't think it attempts to submit.

![Screenshot 2024-08-02 at 9 30 41 AM](https://github.com/user-attachments/assets/7ad587ad-5883-419c-9036-310028830d63)

patrickseguraoddball commented:
Also realized this button needs to be outline instead of filled when adding another participant

<img width="527" alt="Screenshot 2024-08-02 at 10 12 42 AM" src="https://github.com/user-attachments/assets/fd1b69a2-9989-48c0-ac34-c7ac6b9bbd01">

aterstriep commented:
@patrickseguraoddball This is ready for re-review!

I updated the field validation so that there is now a field error message if a user has already been added as a GRB reviewer. The submit button disabled state and style updates are also fixed.

patrickseguraoddball reviewed (approved):
Changes look good, thanks!
